### PR TITLE
chore: logout iframe oauth2

### PR DESCRIPTION
**Pull request by thisyahlen-deriv**

### CODE REVIEW

**Review by yashim-deriv — APPROVED** [review]

File: src/hooks/custom-hooks/useRedirectToOauth.ts [hunk 1]
```diff
@@ -2,15 +2,38 @@ import { useCallback, useEffect, useState } from 'react';
...
 
 const useRedirectToOauth = () => {
     const [shouldRedirect, setShouldRedirect] = useState(false);
     const { error, isAuthorized, isAuthorizing } = useAuthData();
     const isEndpointPage = getCurrentRoute() === 'endpoint';
     const oauthUrl = getOauthUrl();
+
+    const [isOauth2Enabled] = useGrowthbookGetFeatureValue<boolean>({
+        defaultValue: false,
+        featureFlag: 'enableOAuth2',
```
Comment (ali-hosseini-deriv) [inline src/hooks/custom-hooks/useRedirectToOauth.ts:15]:
```suggestion
        featureFlag: 'oauth2_service_enabled',
```

File: src/hooks/custom-hooks/useRedirectToOauth.ts [hunk 2]
```diff
@@ -2,15 +2,38 @@ import { useCallback, useEffect, useState } from 'react';
...
+    });
+
     const redirectToOauth = useCallback(() => {
-        shouldRedirect && window.open(oauthUrl, '_self');
-    }, [oauthUrl, shouldRedirect]);
+        // Testing the use of the iframe element
+        if (shouldRedirect && isOauth2Enabled) {
```
Comment (farrah-deriv) [inline src/hooks/custom-hooks/useRedirectToOauth.ts:20]: what if the feature flag is disabled?

File: src/hooks/custom-hooks/useRedirectToOauth.ts [hunk 3]
```diff
@@ -2,15 +2,38 @@ import { useCallback, useEffect, useState } from 'react';
...
 import { useAuthData } from '@deriv-com/api-hooks';
+import useGrowthbookGetFeatureValue from './useGrowthbookGetFeatureValue';
 
 const useRedirectToOauth = () => {
     const [shouldRedirect, setShouldRedirect] = useState(false);
     const { error, isAuthorized, isAuthorizing } = useAuthData();
     const isEndpointPage = getCurrentRoute() === 'endpoint';
     const oauthUrl = getOauthUrl();
+
+    const [isOauth2Enabled] = useGrowthbookGetFeatureValue<boolean>({
```
Comment (ali-hosseini-deriv) [inline src/hooks/custom-hooks/useRedirectToOauth.ts:13]:
```suggestion
    const [isOauth2Enabled, isGBLoaded] = useGrowthbookGetFeatureValue<boolean>({
```

File: src/hooks/custom-hooks/useRedirectToOauth.ts (hunk 2 above)
Comment (ali-hosseini-deriv) [inline src/hooks/custom-hooks/useRedirectToOauth.ts:20]:
```suggestion
        if (shouldRedirect && isOauth2Enabled && isGBLoaded) {
```

Comment (thisyahlen-deriv) [reply src/hooks/custom-hooks/useRedirectToOauth.ts:20]: Yes missed that. Added it already 🙏

File: src/hooks/custom-hooks/useRedirectToOauth.ts [hunk 4]
```diff
@@ -21,6 +44,10 @@ const useRedirectToOauth = () => {
         }
     }, [error, isAuthorized, isAuthorizing, isEndpointPage, oauthUrl]);
 
+    useEffect(() => {
+        redirectToOauth();
+    }, [redirectToOauth]);
```
Comment (ali-hosseini-deriv) [inline src/hooks/custom-hooks/useRedirectToOauth.ts:51]:
```suggestion
    useEffect(() => {
        if(isGBLoaded)
            redirectToOauth();
    }, [redirectToOauth, isGBLoaded]);
```

**Review by ali-hosseini-deriv — CHANGES_REQUESTED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Review by ali-hosseini-deriv — APPROVED** [review]

**Pull request merged**